### ISSUE TITLE
sentence classification release date?

### ISSUE THREAD

**Issue by jonnykira** [jiesutd/NCRFpp#97]
Hello,

I opened an issue on Jan 24th asking about sentence/document classification functionality to which you replied there would be a release soon. Can you provide any updates on this? possibly an estimated release date?

Thank you

**Comment by jiesutd** [comment]
Hi @jonnykira , thanks for your interest. Currently, the basic classification functions are ready. But more features and test are ongoing. It may take me about another one month to make it ready as I am quite busy these days.

**Comment by chuanfanyoudong** [comment]
Hi, @jiesutd i run a sentence classfication use lstm, and i find it can run and ritch 0.94 of P

**Comment by jiesutd** [comment]
@chuanfanyoudong glad to hear that. Indeed the sentence classification can work properly. But I am trying to add more feature and make the code more concise.

**Comment by chuanfanyoudong** [comment]
@jiesutd 赞

**Comment by jonnykira** [comment]
Hello,

I am also able to get the sentence classification working, just needed to change the format of my training data to sentence, label pairs per line and ensure the delimiter (the `split_token`) between these was correct. From my reading of the code it seems like a character level NN is used to generate a sentence representation which is then concatenated with the word embedding sentence representation, which is then fed to another NN layer followed by crf/softmax to generate the sentence predictions.

**Comment by jiesutd** [comment]
> Hello,
> 
> I am also able to get the sentence classification working, just needed to change the format of my training data to sentence, label pairs per line and ensure the delimiter (the `split_token`) between these was correct. From my reading of the code it seems like a character level NN is used to generate a sentence representation which is then concatenated with the word embedding sentence representation, which is then fed to another NN layer followed by crf/softmax to generate the sentence predictions.

Yes, the sentence classification has the same feature extraction structure as the sequence labeling. The difference is the final output is a sentence level classifier rather than each token. I am still working on make the sentence classification function more powerful and easy-to-use.

**Comment by chuanfanyoudong** [comment]
@jiesutd there is one thing i can't understand, why there id a function called "sentence_representation" and i find  this function is almost the same as the function forward except the "torch.cat(feature_list, 1)" ?

**Comment by aitor-garcia-p** [comment]
> Yes, the sentence classification has the same feature extraction structure as the sequence labeling. The difference is the final output is a sentence level classifier rather than each token. I am still working on make the sentence classification function more powerful and easy-to-use.

Hello,
First of all, congratulations and thanks for sharing this awesome work :-) 
I am also waiting for updates about sentence classification (I made it work but it was a bit difficult to understand).
I have a question about it. Does the sentence classification allow custom features? (like [CAP]=0 [POS]=NP , etc. for sequence labelling). If so, what would be the correct format of the training file?
Thank you :-)

**Comment by jiesutd** [comment]
@aitor-garcia-p Good question! This is what I am working on. The current version does not support the custom features, but I am working on adding this function in. However, I am quite busy in recent days, I can not guarantee when will I release a complete version of the sentence classification.

**Comment by jiesutd** [comment]
> @jiesutd there is one thing i can't understand, why there id a function called "sentence_representation" and i find this function is almost the same as the function forward except the "torch.cat(feature_list, 1)" ?

The sentence classification is to extract the representation of the whole sentence.  They are more difference than the "torch.cat(feature_list, 1)". You can compare them carefully.

**Comment by chuanfanyoudong** [comment]
@jiesutd thank you, I understood wrong before.，i will compare them more carefully

**Comment by shenxuhui** [comment]
Thanks for your work，我是来催更的，哈哈。

**Comment by jiesutd** [comment]
@aitor-garcia-p @jonny-d @shenxuhui 

Hi everyone,  I have just released the sentence classification version in the dev branch: https://github.com/jiesutd/NCRFpp/tree/dev 

Welcome to try it if you are interested. It supported the additional features as @aitor-garcia-p  requested. You can add any features you want (just follow the format as file `sample_data/clf.train.feature.txt`). 

You can start with the demo code `python ncrf.py --config demo.clf.train.config`

There are several major revisions in this branch, I will work on this branch to integrate BERTs and will merge it into the main branch later. 

I just notice that it has been one year since my last response. How time flies! Thank you for your support.

**Comment by aitor-garcia-p** [comment]
Thank you very much for your work :-)
The time flies indeed!
I am terribly busy these days, but if I have the chance I will give it a look.
